python/tests/test_parsimony.py

```diff
@@ -520,16 +533,45 @@ class TestParsimonyBase:
     Base class for tests of the map_mutations parsimony method.
     """
 
-    def do_map_mutations(self, tree, genotypes, alleles=None, compare_lib=True):
+    def do_map_mutations(
+        self, tree, genotypes, alleles=None, ancestral_state=None, compare_lib=True
+    ):
         if alleles is None:
             alleles = [str(j) for j in range(max(genotypes) + 1)]
-        ancestral_state, transitions = hartigan_map_mutations(tree, genotypes, alleles)
         if compare_lib:
-            ancestral_state1, transitions1 = tree.map_mutations(genotypes, alleles)
-            assert ancestral_state == ancestral_state1
-            assert len(transitions) == len(transitions1)
-            assert transitions == transitions1
-        return ancestral_state, transitions
+            # Test the built-in map_mutations first
+            ancestral_state1, transitions1 = tree.map_mutations(
+                genotypes, alleles, ancestral_state
+            )
+            ancestral_state2, transitions2 = hartigan_map_mutations(
+                tree, genotypes, alleles, ancestral_state
+            )
+            assert ancestral_state1 == ancestral_state2
+            assert len(transitions1) == len(transitions2)
+            assert transitions1 == transitions2
+        else:
+            # Test the implementation in this file first
```

Comment:
What's this change for? It's equivalent to what was there before isn't it, just with code duplication?

Reply:
I don't think so. In the previous version, whether or not we specified `compare_lib`, we ran `hartigan_map_mutations` first. Therefore if that function raised an error, we never went on to test whether `tskit.map_mutations` raised the same error. The way I've structured it, if you specify `compare_lib` you test `tskit.map_mutations` first, so can check for the correct error raised there.

Reply:
If we're checking for errors then call the library version directly.

Reply:
Yep, good point.